src/test/java/uk/gov/digital/ho/pttg/application/namematching/NameMatchingCandidatesServiceTest.java

```diff
@@ -9,42 +9,47 @@
 import uk.gov.digital.ho.pttg.application.namematching.candidates.*;
 
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
 import static org.assertj.core.api.Java6Assertions.assertThat;
 import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 @RunWith(MockitoJUnitRunner.class)
 public class NameMatchingCandidatesServiceTest {
 
     private NameMatchingCandidatesService nameMatchingCandidatesService;
 
-    @Mock
-    private NameCombinations nameCombinations;
-    @Mock
-    private MultipleLastNames multipleLastNames;
-    @Mock
-    private SpecialCharacters specialCharacters;
-    @Mock
-    private AliasCombinations aliasCombinations;
-    @Mock
-    private EntireNonAliasName entireNonAliasName;
-    @Mock
-    private EntireLastNameAndEachFirstName entireLastNameAndEachFirstName;
-    @Mock
-    private NamesWithFullStopSpaceCombinations namesWithFullStopSpaceCombinations;
+    @Mock private EntireNonAliasName entireNonAliasName;
+    @Mock private EntireLastNameAndEachFirstName entireLastNameAndEachFirstName;
+    @Mock private MultipleLastNames multipleLastNames;
+    @Mock private NamesWithFullStopSpaceCombinations namesWithFullStopSpaceCombinations;
+    @Mock private AliasCombinations aliasCombinations;
+    @Mock private NameCombinations nameCombinations;
+    @Mock private SpecialCharacters specialCharacters;
 
     @Before
     public void setUp() {
         nameMatchingCandidatesService = new NameMatchingCandidatesService(
-                nameCombinations,
-                multipleLastNames,
-                specialCharacters,
-                aliasCombinations,
-                entireNonAliasName,
-                entireLastNameAndEachFirstName,
-                namesWithFullStopSpaceCombinations);
+                Arrays.asList(
+                        entireNonAliasName,
+                        entireLastNameAndEachFirstName,
+                        multipleLastNames,
+                        namesWithFullStopSpaceCombinations,
+                        aliasCombinations,
+                        nameCombinations,
+                        specialCharacters
+                        ));
+
+        when(entireNonAliasName.generateCandidates(any(InputNames.class))).thenReturn(Collections.emptyList());
```

Comment:
Could we perhaps loop over the list since they're all mocked equally?

Reply:
I did consider that, but I feel it is usually best to be explicit in how test collaborators are initialised.